managers/s3.py

```diff
@@ -37,11 +42,79 @@ def createS3Bucket(self, bucketName, bucketPermissions):
         raise S3Error('Unable to create bucket in s3')
 
 
-    def createManifestInS3(self, manifestPath, manifestJSON, s3_bucket: str):
+    def store_pdf_manifest(self, record, bucket_name, requires_login: bool):
+        for link in record.has_part:
+            item_no, uri, source, media_type, _ = link.split('|')
+
+            if media_type == 'application/pdf':
+                record_id = record.identifiers[0].split('|')[0]
+
+                if requires_login:
+                    manifest_path = f'manifests/publisher_backlist/{source}/{record_id}.json'
+                else:
+                    manifest_path = f'manifests/{source}/{record_id}.json''manifests/{}/{}.json'
+
+                manifest_uri = get_stored_file_url(storage_name=bucket_name, file_path=manifest_path)
+
+                manifest_json = self.generate_manifest(record, uri, manifest_uri)
+
+                self.create_manifest_in_s3(manifest_path, manifest_json, bucket_name)
+
+                if source == 'clacso':
+                    link_string = Part(
+                        index=item_no, 
+                        url=manifest_uri,
+                        source=source,
+                        file_type='application/webpub+json',
+                        flags=json.dumps(dataclasses.asdict(FileFlags()))
+                    ).to_string()
+                elif requires_login:
+                    manifest_flags = {
+                    'catalog': False,
+                    'download': False,
+                    'reader': True,
+                    'embed': False,
+                    **({'fulfill_limited_access': False} if requires_login else {})
+                    }
```

Comment:
I think we could just add a fullfill_limited_access field to the FileFlags data class and set it in the FileFlags.

Reply:
That makes sense